src/passes/strings-encoding/StringEncoding.cpp

```diff
@@ -553,30 +556,56 @@ static Expected<std::unique_ptr<llvm::Module>> loadModule(StringRef Path,
   return M;
 }
 
+static Expected<std::unique_ptr<llvm::Module>>
+generateModule(StringRef Routine, StringRef Triple, LLVMContext &Ctx,
+               ArrayRef<std::string> ExtraArgs) {
+  llvm::hash_code HashValue = llvm::hash_combine(Routine, Triple);
+  const std::string &IRModuleFilename =
+      "/tmp/module_" + std::to_string(HashValue) + ".ll";
+
+  if (llvm::sys::fs::exists(IRModuleFilename)) {
+    return loadModule(IRModuleFilename, Ctx);
+  } else {
+    auto MaybePath = runClangExecutable(Routine, "cpp", Triple, ExtraArgs);
+    if (!MaybePath)
+      return MaybePath.takeError();
+
+    auto MaybeModule = loadModule(*MaybePath, Ctx);
+    if (!MaybeModule)
+      return MaybeModule.takeError();
+
+    std::error_code EC;
+    llvm::raw_fd_ostream IRModuleFile(IRModuleFilename, EC,
+                                      llvm::sys::fs::OF_None);
+    (*MaybeModule)->print(IRModuleFile, nullptr);
```

Comment:
Note: We load the module from disk and write it again with a new name here. For a quick solution that will do. If you want to keep it like that, we may consider a prefix like `omvll-cache-`.

Just for reference, the `runClangExecutable()` function will do something like:
```
$ clang-14 -S -emit-llvm -std=c++17 -o \
/var/folders/2k/myk8kt8d4f52dzr19331wtxr0000gn/T/omvll-x86_64-apple-darwin22.4.0-78bae9.cpp.ll \
/var/folders/2k/myk8kt8d4f52dzr19331wtxr0000gn/T/omvll-x86_64-apple-darwin22.4.0-78bae9.cpp
```

Reply:
Should resemble this now, except that instead of using the placeholder `%%%%%%` needed when using `createUniqueFile`, we use our hash value after retrieving system tmp dir. What do you think?